File: engine/admin/model/process_detail_response.py
#!/usr/bin/python3

from datetime import datetime
from process_pool.process import ProcessState


class ProcessDetailResponse:
  '''
  Represents the response returned by admin's ping.
  Contains all information and details about the ongoing state of a process

  Attributes
    id (int)                  - Unique identifier of a process
    state (ProcessState)      - State of the process
    flip_flop_count (int)     - Count of times the status was bounced
    start_datetime (datetime) - datetime of when the process was started
  '''


  def __init__(self, id: int, state: ProcessState, flip_flop_count: int, start_datetime: datetime):
    self.id = id
    self.state = state
    self.flip_flop_count = flip_flop_count
    self.start_datetime = start_datetime


  def get_id(self) -> int:
    return self.id


  def get_state(self) -> ProcessState:
    return self.state


  def get_flip_flop_count(self) -> int:
    return self.flip_flop_count


  def get_start_datetime(self) -> datetime:
    return self.start_datetime


  def __str__(self) -> str:
    return str(self.to_json())


  def __repr__(self) -> str:
    return self.__str__()


  def __eq__(self, other) -> bool:
    return isinstance(other, ProcessDetailResponse) and\
      self.id == other.id and\
      self.state == other.state and\
      self.flip_flop_count == other.flip_flop_count and\
      self.start_datetime == other.start_datetime


  def to_json(self) -> dict:
    return dict(
      id=self.id,
      state=self.state.name,
      flipFlopCount=self.flip_flop_count,
      startDatetime=str(self.start_datetime)
    )

File: service/model/src/main/java/org/mlsk/service/model/admin/ProcessDetailResponse.java
package org.mlsk.service.model.admin;

import java.util.Objects;

public class ProcessDetailResponse {

  private final int id;
  private final String state;
  private final long flipFlopCount;
  private final String startDatetime;

  public ProcessDetailResponse(int id, String state, long flipFlopCount, String startDatetime) {
    this.id = id;
    this.state = state;
    this.flipFlopCount = flipFlopCount;
    this.startDatetime = startDatetime;
  }

  public int getId() {
    return id;
  }

  public String getState() {
    return state;
  }

  public long getFlipFlopCount() {
    return flipFlopCount;
  }

  public String getStartDatetime() {
    return startDatetime;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    ProcessDetailResponse that = (ProcessDetailResponse) o;
    return id == that.id && flipFlopCount == that.flipFlopCount && Objects.equals(state, that.state) && Objects.equals(startDatetime, that.startDatetime);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, state, flipFlopCount, startDatetime);
  }

  @Override
  public String toString() {
    return "ProcessDetailResponse{" +
        "id=" + id +
        ", state='" + state + '\'' +
        ", flipFlopCount=" + flipFlopCount +
        ", startDatetime='" + startDatetime + '\'' +
        '}';
  }
}
